azure-kusto-data/src/client.rs

```diff
@@ -104,137 +116,170 @@ impl KustoClient {
+    ///
+    ///   for table in result.into_primary_results() {
+    ///        println!("{}", table.table_name);
+    ///    }
+    /// # Ok(())}
+    /// ```
+    ///
+    #[must_use]
+    pub fn execute_query_with_options(
         &self,
-        database: DB,
-        query: Q,
+        database: String,
+        query: String,
         options: Option<RequestOptions>,
-    ) -> V2QueryRunner
-    where
-        DB: Into<String>,
-        Q: Into<String>,
-    {
+    ) -> V2QueryRunner {
         V2QueryRunner(self.execute_with_options(database, query, QueryKind::Query, options))
     }
 
-    pub fn execute_query<DB, Q>(&self, database: DB, query: Q) -> V2QueryRunner
-    where
-        DB: Into<String>,
-        Q: Into<String>,
-    {
+    /// Execute a KQL query.
+    /// To learn more about KQL go to [https://docs.microsoft.com/en-us/azure/kusto/query/](https://docs.microsoft.com/en-us/azure/kusto/query)
+    ///
+    /// # Example
+    /// ```no_run
+    /// use azure_kusto_data::prelude::*;
+    ///
+    /// # #[tokio::main] async fn main() -> Result<(), Error> {
+    /// let client = KustoClient::new(
+    ///    ConnectionString::with_default_auth("https://mycluster.region.kusto.windows.net/".to_string()),
+    ///    KustoClientOptions::default())?;
+    ///
+    ///   // Once the [IntoFuture] trait is stabilized, we can drop the call the `into_future()` here
+    ///    let result = client.execute_query("some_database".into(), "MyTable | take 10".into()).into_future().await?;
+    ///
+    ///    for table in result.into_primary_results() {
+    ///        println!("{}", table.table_name);
+    ///    }
+    /// # Ok(())}
+    /// ```
+    #[must_use]
+    pub fn execute_query(&self, database: String, query: String) -> V2QueryRunner {
         V2QueryRunner(self.execute_with_options(database, query, QueryKind::Query, None))
     }
 
-    pub fn execute_command_with_options<DB, Q>(
+    /// Execute a management command with additional options.
+    /// To learn more about see [commands](https://docs.microsoft.com/en-us/azure/data-explorer/kusto/management/)
+    ///
+    /// # Example
+    /// ```no_run
+    /// use azure_kusto_data::prelude::*;
+    /// # #[tokio::main] async fn main() -> Result<(), Error> {
+    /// let client = KustoClient::new(
+    ///    ConnectionString::with_default_auth("https://mycluster.region.kusto.windows.net/".to_string()),
+    ///    KustoClientOptions::default())?;
+    ///
+    /// // Once the [IntoFuture] trait is stabilized, we can drop the call the `into_future()` here
+    ///    let result = client.execute_command_with_options("some_database".into(), ".show version".into(),
+    ///     Some(RequestOptionsBuilder::default().with_request_app_name("app name".to_string()).build().unwrap()))
+    ///     .into_future().await?;
+    ///
+    /// for table in result.tables {
+    ///        println!("{}", table.table_name);
+    ///    }
+    /// # Ok(())}
+    /// ```
+    #[must_use]
```

Comment:
Right now must_use probably makes sense here, but I was thinking abut the target state a bit ... if I remember correctly, we want to handle errors returned from kusto more explicitly, rather then having the user inspect the result tables for that.

If we had such error reporting already, would we still make this must use? With commands I suspect I often just want to know its executed without errors, and if so, an not necessarily interested in details - i.e. handle the erros, but the can be done without using the result?

Reply:
In the end, we would probably always want to return a future (right?), and a future does nothing until it's awaited or sent to the loop, so the user will always need to use it, even if we'll get errors before.

Reply:
I think I misunderstood what this means, and essentially thought you would have to assign it to something. But consuming it is using it :). The following would work as it consumes the future, right?

```rs
client.execute_command(...).await?;
```